packages/graph/hash_graph/lib/graph/src/api/web/mod.rs

```diff
@@ -0,0 +1,99 @@
+//! The Axum webserver for accessing the Graph API datastore operations.
+//!
+//! Handler methods are grouped by routes that make up the `web_api`. See [`data_type`] for a
+//! grouped router example
+
+mod api_resource;
+mod data_type;
+mod error;
+mod property_type;
+
+use axum::{routing::get, Extension, Json, Router};
+use utoipa::{openapi, Modify, OpenApi};
+
+use self::api_resource::RoutedResource;
+use crate::datastore::Datastore;
+
+fn api_resources<T: Datastore>() -> Vec<Router> {
```

Comment:
I would really like to combine this and the `api_documentation` functions in one, and only specify endpoints once, but the `RoutedResource` cannot be turned into a trait object. I would appreciate some tips/alternatives. The generic is what makes it difficult, as we don't have control over the generated implementations of `utoipa`, and thus are unable to pass the generic through there (although we could implement OpenApi by hand and get around that issue..)